client/src/cmdhfmfdes.c

```diff
@@ -4433,6 +4477,17 @@ static int CmdHF14aDesChk(const char *Cmd) {
     uint8_t app_ids[78] = {0};
     uint32_t app_ids_len = 0;
 
+    clearCommandBuffer();
+
+    mfdes_info_res_t info = {0};
+    res = mfdes_get_info(&info);
+    if (res != PM3_SUCCESS) {
+        return res;
+    }
+    // TODO: Store this UID someowhere not global
+    memcpy(tag->info.uid, info.uid, info.uidlen);
+    tag->info.uidlen = info.uidlen;
+
```

Comment:
I had to retrieve the card's UID here as it's not stored with the client in any global/local variable. I'm sure there's a better way to do this. but it works for now. though I do occasionally get a _Failed to Select card_